### PR TITLE
Add support for shutdown API and add client docs

### PR DESCRIPTION
**Pull request by halspang**
# Description

This commit adds support for the shutdown API. It also adds docs for
that method and a few others which were missing from the docs.

https://github.com/dapr/dotnet-sdk/issues/914

Signed-off-by: Hal Spang <halspang@microsoft.com>

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #914 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

File: src/Dapr.Client/DaprClientGrpc.cs [hunk 1]
```diff
@@ -1356,7 +1313,63 @@ private Autogenerated.StateItem ToAutogeneratedStateItem(StateTransactionRequest
...
+        }
+
+        /// <inheritdoc/>
+        public override async Task<bool> CheckOutboundHealthAsync(CancellationToken cancellationToken = default)
+        {
+            var path = "/v1.0/healthz/outbound";
+            var request = new HttpRequestMessage(HttpMethod.Get, new Uri(this.httpEndpoint, path));
+            try
+            {
+                var response = await this.httpClient.SendAsync(request, cancellationToken);
```
Comment (mcandeia) [inline src/Dapr.Client/DaprClientGrpc.cs:1343]: nitpicking:

This is a really lightweight request but given that you don't even need the request content you can ignore it by adding `HttpCompletionOption.ResponseHeadersRead`
```suggestion
                using var response = await this.httpClient.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, cancellationToken);
```

File: src/Dapr.Client/DaprClientGrpc.cs [hunk 2]
```diff
@@ -1356,7 +1313,63 @@ private Autogenerated.StateItem ToAutogeneratedStateItem(StateTransactionRequest
...
+        #region Dapr Sidecar Methods
+
+        /// <inheritdoc/>
+        public override async Task<bool> CheckHealthAsync(CancellationToken cancellationToken = default)
+        {
+            var path = "/v1.0/healthz";
+            var request = new HttpRequestMessage(HttpMethod.Get, new Uri(this.httpEndpoint, path));
+            try
+            {
+                var response = await this.httpClient.SendAsync(request, cancellationToken);
```
Comment (mcandeia) [inline src/Dapr.Client/DaprClientGrpc.cs:1327]: nitpicking:

```suggestion
                using var response = await this.httpClient.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, cancellationToken);
```

File: src/Dapr.Client/DaprClient.cs [hunk 3]
```diff
@@ -330,6 +330,13 @@ public HttpRequestMessage CreateInvokeMethodRequest<TRequest>(string appId, stri
         /// <param name="cancellationToken">A <see cref="CancellationToken" /> that can be used to cancel the operation.</param>
         /// <returns>A <see cref="Task" /> that will return when the operation has completed.</returns>
         public abstract Task WaitForSidecarAsync(CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Send a command to the Dapr Sidecar telling it to shutdown.
+        /// </summary>
+        /// <param name="cancellationToken">A <see cref="CancellationToken" /> that can be used to cancel the operation.</param>
+        /// <returns>A <see cref="Task" /> that will return when the operation has completed.</returns>
+        public abstract Task ShutdownSidecarAsync(CancellationToken cancellationToken = default);
```
[the remark this answers is not in the document]
Comment (halspang) [reply src/Dapr.Client/DaprClient.cs:339]: This is something that we'll define in the sig-sdk group, but I do think it's important to keep naming consistent. I do worry that both of those are a little to lax on the name though. 

Do you think it's worth being different here for clarity? Having the API just be Shutdown, to me, doesn't really tell the user what it's doing.

[the remark this answers is not in the document]
Comment (mcandeia) [reply src/Dapr.Client/DaprClient.cs:339]: I don't have a strong opinion here,

There are negative and positive effects of being explicit in this case, I do not feel much comfortable exposing the word `Sidecar` I would say that "ShutdownDaprd" sound better, and, if by any chance we would have more than one sidecar  (probably won't happen) it becomes explicit which one is shutting down. But again, it's nitpicking and I agree with you that just `Shutdown` is too generic

[the remark this answers is not in the document]
Comment (rynowak) [reply src/Dapr.Client/DaprClient.cs:339]: +1 to everything said here. This seems like a good chance to be consistent on the terminology.

File: src/Dapr.Client/DaprClientGrpc.cs [hunk 4]
```diff
@@ -1356,7 +1313,63 @@ private Autogenerated.StateItem ToAutogeneratedStateItem(StateTransactionRequest
...
+            }
+        }
+
+        /// <inheritdoc/>
+        public async override Task ShutdownSidecarAsync(CancellationToken cancellationToken = default)
+        {
+            await client.ShutdownAsync(new Empty(), CreateCallOptions(null, cancellationToken));
+        }
+
+        #endregion
```
Comment (rynowak) [inline src/Dapr.Client/DaprClientGrpc.cs:1372]: Pretty simple :)

**Review by rynowak — APPROVED** [review]

**Pull request merged**